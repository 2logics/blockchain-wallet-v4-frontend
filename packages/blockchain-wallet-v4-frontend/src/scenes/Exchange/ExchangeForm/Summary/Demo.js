import React from 'react'
import styled from 'styled-components'
import { FormattedMessage } from 'react-intl'

import { Text } from 'blockchain-info-components'
import { Wrapper, Title, Note } from 'components/Exchange'

const DemoWrapper = styled(Wrapper)`
  background-color: #f5f6f8;
`
const DemoTitle = styled(Title)`
  font-size: 24px;
  line-height: 29px;
  color: ${props => props.theme['brand-primary']};
`
const DemoText = styled(Text)`
  font-size: 14px;
  line-height: 24px;
  margin-bottom: 24px;
  color: ${props => props.theme['brand-primary']};
`
const DemoNote = styled(Note)`
  margin-top: 0;
  margin-bottom: 16px;
`
const DemoNoteTitle = styled(Text)`
  display: inline;
  font-size: 12px;
  color: ${props => props.theme['brand-primary']};
`

const DemoSummary = () => (
  <DemoWrapper>
    <DemoTitle>
      <FormattedMessage
        id='scenes.exchange.exchangeform.summary.demo_title'
        defaultMessage="Trading your crypto doesn't mean trading away control."
      />
    </DemoTitle>
    <DemoText>
      <FormattedMessage
        id='scenes.exchange.exchangeform.summary.demo_text'
        defaultMessage='Swap enables you to trade crypto with low fees, live rates, and quick settlement, all while maintaining full control of your funds.'
      />
    </DemoText>
    <DemoNote>
      <DemoNoteTitle>
        <FormattedMessage
          id='scenes.exchange.exchangeform.summary.low_fees'
          defaultMessage='Low Fees'
        />
        {': '}
      </DemoNoteTitle>
      <FormattedMessage
        id='scenes.exchange.exchangeform.summary.demo_note_1'
        defaultMessage="Great news! You don't have to go to an exchange to get the best prices."
      />
    </DemoNote>
    <DemoNote>
      <DemoNoteTitle>
        <FormattedMessage
          id='scenes.exchange.exchangeform.summary.higher_limits'
          defaultMessage='Higher Limits'
        />
        {': '}
      </DemoNoteTitle>
      <FormattedMessage
        id='scenes.exchange.exchangeform.summary.demo_note_2'
        defaultMessage='Like to trade big? Get access to limits of up to $25,000 per day.'
      />
    </DemoNote>
    <DemoNote>
      <DemoNoteTitle>
        <FormattedMessage
          id='scenes.exchange.exchangeform.summary.easy_to_use'
          defaultMessage='Easy to use'
        />
        {': '}
      </DemoNoteTitle>
      <FormattedMessage
        id='scenes.exchange.exchangeform.summary.demo_note_3'
        defaultMessage='Get started in just a few steps'
      />
    </DemoNote>
  </DemoWrapper>
)

export default DemoSummary
