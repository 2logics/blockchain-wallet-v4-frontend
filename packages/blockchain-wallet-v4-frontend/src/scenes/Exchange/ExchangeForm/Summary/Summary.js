import React from 'react'
import { FormattedMessage } from 'react-intl'

import { Wrapper as BorderWrapper, Title, Note } from 'components/Exchange'
import Volume from './Volume'
import RatesBox from '../RatesBox'

export default class Summary extends React.PureComponent {
  render () {
    const { showDemoSummary, sourceCoin, targetCoin, currency } = this.props
    return showDemoSummary ? (
      ''
    ) : (
      <BorderWrapper>
        <Title>
          <FormattedMessage
            id='scenes.exchange.exchangeform.summary.title'
            defaultMessage='Summary'
          />
        </Title>
        <Volume
          sourceCoin={sourceCoin}
          targetCoin={targetCoin}
          currency={currency}
        />
        <Note>
          <FormattedMessage
            id='scenes.exchange.exchangeform.summary.note'
            defaultMessage='All amounts are correct at this time but may change depending on the market price and network congestion at the time of your transaction.'
          />
        </Note>
        <RatesBox
          sourceCoin={sourceCoin}
          targetCoin={targetCoin}
          currency={currency}
        />
      </BorderWrapper>
    )
  }
}
