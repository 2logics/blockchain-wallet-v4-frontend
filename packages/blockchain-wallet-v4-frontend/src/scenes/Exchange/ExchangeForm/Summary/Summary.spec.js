import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Summary from './Summary'

const props = {
  sourceCoin: 'BTC',
  targetCoin: 'ETH',
  currency: '$'
}

describe('Exchange Summary', () => {
  it(`should render demo summary correctly`, () => {
    const component = shallow(<Summary {...props} showDemoSummary />)

    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
  it(`should render summary correctly`, () => {
    const component = shallow(<Summary {...props} showDemoSummary={false} />)

    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
