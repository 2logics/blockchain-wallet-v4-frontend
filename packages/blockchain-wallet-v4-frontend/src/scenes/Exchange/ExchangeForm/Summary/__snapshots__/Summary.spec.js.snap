// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange Summary should render demo summary correctly 1`] = `<DemoSummary />`;

exports[`Exchange Summary should render summary correctly 1`] = `
<Exchange__Wrapper>
  <Exchange__Title>
    <FormattedMessage
      defaultMessage="Summary"
      id="scenes.exchange.exchangeform.summary.title"
      values={Object {}}
    />
  </Exchange__Title>
  <Connect(Volume)
    currency="$"
    sourceCoin="BTC"
    targetCoin="ETH"
  />
  <Exchange__Note>
    <FormattedMessage
      defaultMessage="All amounts are correct at this time but may change depending on the market price and network congestion at the time of your transaction."
      id="scenes.exchange.exchangeform.summary.note"
      values={Object {}}
    />
  </Exchange__Note>
  <Connect(RatesBox)
    currency="$"
    sourceCoin="BTC"
    targetCoin="ETH"
  />
</Exchange__Wrapper>
`;
